### DISCUSSION TITLE
Drawing shapes

### DISCUSSION THREAD

**Discussion by aagha** [tannerhelland/PhotoDemon#427] [General]
I'm not sure if I'm missing something obvious or of the feature is missing, but under "Paint", I'd expect a menu item to be able to draw a shape.

I want to draw a red circle, draw an arrow to it, and write some text.

Is there an ability to draw circles, shapes, arrows?

For example, if you look a the red circle in https://github.com/tannerhelland/PhotoDemon/issues/285#issuecomment-573343045, can that be done w/ PD?

**Comment by tannerhelland** [comment]
Hello @aagha!  Thank you for trying PhotoDemon.

I am happy (and sad?) to say that you are not missing anything. 😄 PhotoDemon does not currently provide a shape tool.  

I had hoped to ship a shape tool in version 9.0 (due out in September), and even added SVG support in preparation for this, but I wasn't able to complete it in time.  I apologize for this.  I now hope to have the shape tool ready sometime later this year.

In the meantime, if you need a full-featured shape tool, I would probably suggest using Paint.NET.  

You *can* use a "hack" to create shapes in PhotoDemon by using any of the Selection tools, then using the Edit > Fill or Edit > Stroke menus to fill or outline the shape, respectively.  (Or the identical Select > Fill and Select > Stroke menus.)  This is how I created the circle you mention in #285.  

Here is a screen capture of the process:

![select-tool-shapes-hack](https://user-images.githubusercontent.com/1930029/187550185-e6214c09-887c-4dcc-b976-e0c85a926406.png)

(This solution is far from ideal, I know.)

**Reply by aagha** [reply]
Ah!

Thanks for the response @tannerhelland.

First off--nothing to be sorry about. Your software is a marvel and piece of art.

I'll keep an eye out for a future version with shape support!

**Comment by aagha** [comment]
Hi! I have note to come back and re-visit this.  @tannerhelland Did a markup tool get added yet? If not, I'll check back in 6 mos.

**Reply by tannerhelland** [reply]
Hello @aagha !  Thank you for checking back on this.

Unfortunately, no - PhotoDemon does not have this tool yet.  

I hope to have good news six months from now.  Please return to keep me honest.  🙏

**Reply by aagha** [reply]
Hi @tannerhelland -- I thought I'd follow up on my last question and see if you've had a chance to add this feature yet.

**Reply by tannerhelland** [reply]
Thank you so much for checking in, @aagha .  I'm afraid this feature still is not ready!  I'm so sorry!

**Reply by Edinho367** [reply]
Hello, is there any markup tool, or an 'hack' to get similar result?

I really enjoyed the hack you showed up for drawing shapes, it's enough when we need a basic shape...

**Reply by hi5** [reply]
If by markup you mean "highlighting with a marker":

1. You can do it directly on the same layer if you choose a highlight color, say yellow, brush tool,  and then select blend mode color (or hue, there are various options to try) -- see screenshot below
2. You can do the same on a new layer and then just play with the layer blend mode, that way you can "edit" the highlights too and if keep the layers in your source file they can be turned on/off 

![markup](https://github.com/user-attachments/assets/3708e6b4-f76c-47ef-bc9a-f7f3e2dfe325)

**Reply by Edinho367** [reply]
Thanks! I tried, and it worked even better with selection tool for delimitation of the brush (making the highlight straight)
Very nice!

**Comment by Vlada3333** [comment]
thanks for the information!

**Comment by alanbork** [comment]
still no shape tool? or is it just well hidden? Being able to add an arrow is a major part of my workflow.

**Reply by Edinho367** [reply]
Yeah, i second that it would be great if we had some shapes, like straight and curved (with custom number of nodes) lines and arrows, squares and circles to make it one color outline and another color interior...

**Reply by alanbork** [reply]
for sure, but that's a big wish list and could take a long time. Just being able to add an arrow would be a HUGE improvement.  We don't really need a full vector art package as inkscape etc exist.

**Comment by Edinho367** [comment]
Hey DEV, I just wanted to share a quick idea!

_(Just a heads up, I have absolutely zero programming knowledge.)_

About the need for basic shape tools, I'd point out an open-source screen capture tool called **ShareX** that has really robust editing capabilities (although lacks a lot of PhotoDemon features)

Perhaps you could check out their implementation. It might serve as a shortcut or a good source of inspiration, potentially making it way easier than building the functionality completely from scratch.

Thanks a lot for your hard work on PhotoDemon!

ShareX GitHub Link: [https://github.com/ShareX/ShareX](https://github.com/ShareX/ShareX)
